packages/zipkin-transport-aws-sqs/src/AwsSqsLogger.js

```diff
@@ -0,0 +1,89 @@
+const AWS = require('aws-sdk');
+const {jsonEncoder: {JSON_V2}} = require('zipkin');
+const {EventEmitter} = require('events');
+
+class AwsSqsLogger extends EventEmitter {
+  constructor(options) {
+    /* eslint-disable no-console */
+    super();
+    this.log = options.log || console;
+    this.delaySeconds = options.delaySeconds || 0;
+    this.pollerSeconds = options.pollerSeconds || 100;
+    this.queue = [];
+    this.queueBytes = 0;
+    this.errorListenerSet = false;
+    this.maxPayloadSize = 0;
+    this.queueUrl = options.queueUrl;
+    if (typeof options.encoder === 'undefined' || options.encoder === JSON_V2) {
+      this.encoder = JSON_V2;
+    } else {
+      throw new Error('Unsupported encoder. Only support JSON_V2.');
+    }
+    if (typeof options.awsConfig !== 'undefined') {
+      AWS.config.update(options.awsConfig);
+    }
+    this.awsClient = new AWS.SQS({apiVersion: '2012-11-05'});
+
+    const timer = setInterval(() => {
+      this.processQueue();
+    }, this.pollerSeconds);
+    if (timer.unref) { // unref might not be available in browsers
+      timer.unref(); // Allows Node to terminate instead of blocking on timer
+    }
+  }
+
+  _getPayloadSize(nextSpan) {
```

Comment:
`encodedSpan` instead of `nextSpan` I think, it's hard to see this is serialized